pkg/server/server.go

```diff
@@ -542,4 +552,25 @@ func (s *ProxyServer) serveRecvBackend(stream agent.AgentService_ConnectServer,
 		}
 	}
 	klog.Infof("<<< Close backend %v of agent %s", stream, agentID)
+
+	// Close all connected frontends when the agent connection is closed
+	// TODO(#126): Frontends in PendingDial state that have not been added to the
+	//             list of frontends should also be closed.
+	frontends, _ := s.getFrontends(agentID)
```

Comment:
This clean up block should really be turned into a function and called via a defer at the beginning of serveRecvFrontend.

Reply:
I wrapped it in an inline function and put it in a defer